Tasks/Common/azure-arm-rest/azure-arm-compute.ts

```diff
@@ -386,7 +386,60 @@ export class VirtualMachineExtensions {
         this.client = client;
     }
 
-    public get(resourceGroupName, vmName, vmExtensionName, options, callback) {
+    public list(resourceGroupName, resourceName, resourceType, options, callback: azureServiceClient.ApiCallback) {
+        if (!callback && typeof options === 'function') {
+            callback = options;
+            options = null;
+        }
+        if (!callback) {
+            throw new Error(tl.loc("CallbackCannotBeNull"));
+        }
+
+        // Validate
+        try {
+            this.client.isValidResourceGroupName(resourceGroupName);
+            if (resourceName === null || resourceName === undefined || typeof resourceName.valueOf() !== 'string') {
+                throw new Error(tl.loc("ResourceNameCannotBeNull"));
+            }
+        } catch (error) {
+            return callback(error);
+        }
+
+        var httpRequest = new azureServiceClient.WebRequest();
+        httpRequest.method = 'GET';
+        httpRequest.headers = this.client.setCustomHeaders(options);
+        httpRequest.uri = this.client.getRequestUri('//subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Compute/{resourceType}/{resourceName}/extensions',
```

Comment:
Should the strings "virtualMachines" and "virtualMachineScaleSets" be local to methods in these files only, rather than each of their callers?